.github/workflows/sdk-generation.yaml

```diff
@@ -194,17 +266,19 @@ jobs:
     if: ${{ needs.generate.outputs.java_regenerated == 'true' && inputs.publish_java == 'true' && inputs.mode != 'pr' }}
     name: Publish Java SDK
     runs-on: ubuntu-latest
-    needs: generate
+    needs: [generate, compile-java]
     defaults:
       run:
         working-directory: ${{ needs.generate.outputs.java_directory }}
     steps:
       - uses: actions/checkout@v3
+        with:
```

Comment:
@chase-crumbaugh this fixes a bug where it was publishing the wrong commit (generally the last one and not the newly created one)